backend/internal/application/service/applicationservice.go

```diff
@@ -233,6 +241,27 @@ func validateAuthFlowGraphID(app *model.Application) error {
 	return nil
 }
 
+// validateRegistrationFlowGraphID validates the registration flow graph ID for the application.
+// If the graph ID is not provided, it attempts to infer it from the auth flow graph ID.
+func validateRegistrationFlowGraphID(app *model.Application) error {
+	if app.RegistrationFlowGraphID != "" {
+		isValidFlowGraphID := graphservice.GetGraphService().IsValidGraphID(app.RegistrationFlowGraphID)
+		if !isValidFlowGraphID {
+			return fmt.Errorf("invalid registration flow graph ID: %s", app.RegistrationFlowGraphID)
+		}
+	} else {
+		if strings.HasPrefix(app.AuthFlowGraphID, constants.AuthFlowGraphPrefix) {
+			suffix := strings.TrimPrefix(app.AuthFlowGraphID, constants.AuthFlowGraphPrefix)
+			app.RegistrationFlowGraphID = constants.RegistrationFlowGraphPrefix + suffix
```

Comment:
Where does this inferred flow gets build?

Reply:
It's not included in this PR. Will be added with the next PR.

Reply:
Added with https://github.com/asgardeo/thunder/pull/191